internal/services/telegramservice.go

```diff
@@ -0,0 +1,138 @@
+	}
+	defer resp.Body.Close()
+
+	bodyJson, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, utils.CommonError(err.Error() + " " + requestUrl)
+	}
+
+	if utils.IsFailStatus(resp.StatusCode) {
+		return nil, utils.HttpError(resp.StatusCode, resp.Status+" "+requestUrl+" "+string(bodyJson))
+	}
+
+	var getUpdateResponse models.GetUpdateResponse
+	err = json.Unmarshal(bodyJson, &getUpdateResponse)
+	if err != nil {
+		return nil, utils.CommonError(err.Error() + " " + requestUrl)
+	}
+
+	return getUpdateResponse.Result, nil
+}
+
+func SendTextMessage(token string, chatId int, text string) error {
+	requestUrl := utils.TelegramApiUrl + token + "/sendMessage"
+
+	message := models.BotMessage{
+		ChatId:      chatId,
+		Text:        text,
+		ReplyMarkup: models.ReplyKeyboardRemove{RemoveKeyboard: true},
+	}
+
+	messageJson, err := json.Marshal(message)
+	if err != nil {
+		return utils.CommonError(err.Error() + requestUrl + " " + fmt.Sprintf("%v", message))
+	}
+
+	resp, err := http.Post(requestUrl, "application/json", bytes.NewBuffer(messageJson))
+	if err != nil {
+		return utils.CommonError(err.Error() + requestUrl + " " + fmt.Sprintf("%v", message))
+	}
+	if utils.IsFailStatus(resp.StatusCode) {
+		return utils.HttpError(resp.StatusCode, resp.Status+" "+requestUrl+" "+fmt.Sprintf("%v", message)+" "+string(messageJson))
+	}
+
+	return nil
+}
+
+func SendTextButtons(token string, chatId int, text string, textList []string) error {
+	requestUrl := utils.TelegramApiUrl + token + "/sendMessage"
+
+	buttons := make([][]models.KeyboardButton, len(textList))
+	keyboard := models.ReplyKeyboardMarkup{
+		Keyboard: buttons,
+	}
+	for i := range textList {
+		button := models.KeyboardButton{
+			TextButton:      textList[i],
+			RequestLocation: false,
+		}
+		buttons[i] = make([]models.KeyboardButton, 1)
+		buttons[i][0] = button
+	}
+
+	message := models.BotMessage{
+		ChatId:      chatId,
+		Text:        text,
+		ReplyMarkup: keyboard,
+	}
+
+	messageJson, err := json.Marshal(message)
+	if err != nil {
+		return utils.CommonError(err.Error() + requestUrl + " " + fmt.Sprintf("%v", message))
+	}
+
+	resp, err := http.Post(requestUrl, "application/json", bytes.NewBuffer(messageJson))
+	if err != nil {
+		return utils.CommonError(err.Error() + requestUrl + " " + fmt.Sprintf("%v", message))
+	}
+	if utils.IsFailStatus(resp.StatusCode) {
+		return utils.HttpError(resp.StatusCode, resp.Status+" "+requestUrl+" "+fmt.Sprintf("%v", message)+" "+string(messageJson))
+	}
+
+	return nil
+}
+
+func SendLocatonRequest(token string, chatId int, text string) error {
+	requestUrl := utils.TelegramApiUrl + token + "/sendMessage"
+
+	buttons := make([][]models.KeyboardButton, 1)
+	keyboard := models.ReplyKeyboardMarkup{
+		Keyboard: buttons,
+	}
+	button := models.KeyboardButton{
+		TextButton:      "Отправить геолокацию",
+		RequestLocation: true,
+	}
+	buttons[0] = make([]models.KeyboardButton, 1)
+	buttons[0][0] = button
```

Comment:
Можно и проще:

```suggestion
	button := models.KeyboardButton{
		TextButton:      "Отправить геолокацию",
		RequestLocation: true,
	}
	buttons := [][]models.KeyboardButton{
		[]models.KeyboardButton{button},
	}
```

Или даже 

```suggestion
	button := models.KeyboardButton{
		TextButton:      "Отправить геолокацию",
		RequestLocation: true,
	}
	buttons := [][]models.KeyboardButton{
		[]models.KeyboardButton{
			{
				TextButton:      "Отправить геолокацию",
				RequestLocation: true,
			}
		},
	}
```

Такой вариант не ошибка, но просто так не делают обычно, когда нужно константное число элементов вставить.

Reply:
замечательно, спасибо)